src/libexpr/eval.cc

```diff
@@ -2508,7 +2477,8 @@ void EvalState::printStats()
                     else
                         obj.attr("name", nullptr);
                     if (auto pos = positions[fun->pos]) {
-                        obj.attr("file", (std::string_view) pos.file);
+                        if (auto path = std::get_if<SourcePath>(&pos.origin))
```

Comment:
I think it would be nice to cover the other variants (or at least stdin and string) in some way here too -- maybe even add some further metadata to allow distinguishing `--apply` from `--expr` strings. Nothing that needs to be resolved in this PR though IMHO.